src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -0,0 +1,420 @@
+
+            try {
+                $command = new EditRootCategoryCommand($categoryId);
+
+                $this->populateCommandWithFormData($command, $data);
+
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_category_add');
+            } catch (CategoryException $e) {
+                $this->addFlash('error', $this->handleEditException($e));
+            }
+        }
+
+        /** @var DefaultGroups $defaultGroups */
+        $defaultGroups = $this->getQueryBus()->handle(new GetDefaultGroups());
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Categories/edit_root.html.twig', [
+            'layoutTitle' => $this->trans(
+                'Edit: %value%',
+                'Admin.Catalog.Feature',
+                [
+                    '%value%' => $editableCategory->getName()[$this->getContextLangId()],
+                ]
+            ),
+            'editRootCategoryForm' => $rootCategoryForm->createView(),
+            'editableCategory' => $editableCategory,
+            'defaultGroups' => $defaultGroups,
+        ]);
+    }
+
+    /**
+     * @param AbstractRootCategoryCommand $command
+     * @param array $data
+     */
+    protected function populateCommandWithFormData(AbstractRootCategoryCommand $command, array $data)
+    {
+        if (null !== $data['description']) {
+            $command->setDescription($data['description']);
+        }
+
+        if (null !== $data['meta_title']) {
+            $command->setMetaTitle($data['meta_title']);
+        }
+
+        if (null !== $data['meta_description']) {
+            $command->setMetaDescription($data['meta_description']);
+        }
+
+        if (null !== $data['meta_keyword']) {
+            $command->setMetaKeywords($data['meta_keyword']);
+        }
+
+        if (null !== $data['group_association']) {
+            $command->setAssociatedGroupIds($data['group_association']);
+        }
+
+        if (null !== $data['shop_association']) {
+            $command->setAssociatedShopIds($data['shop_association']);
+        }
+
+        if (null !== $data['cover_image']) {
+            $command->setCoverImage($data['cover_image']);
+        }
+
+        if (null !== $data['thumbnail_image']) {
+            $command->setThumbnailImage($data['thumbnail_image']);
+        }
+
+        if (null !== $data['menu_thumbnail_images']) {
+            $command->setMenuThumbnailImages($data['menu_thumbnail_images']);
+        }
+    }
+
+    /**
+     * @param CategoryException $exception
+     *
+     * @return string User friendly error message for exception
+     */
+    protected function handleAddException(CategoryException $exception)
+    {
+        $type = get_class($exception);
+
+        if (CategoryConstraintException::class === $type) {
+            return $this->handleConstraintException($exception);
+        }
+
+        $messages = [
```

Comment:
Can we rename it `availableMessagesForDisplay` or something like that ? It took me a few minutes to understand the `if (isset($messages[$type]))` logic so maybe we can make it a bit more understandable